R/sass_cache.R

```diff
@@ -0,0 +1,148 @@
+#' used instead of performing the compilation. If it does not exist, then
+#' compilation is performed and usual and the results are stored in the cache.
+#'
+#' If a file that is included using [sass_file()] changes on disk (i.e. its
+#' last-modified time changes), its previous cache entries will effectively be
+#' invalidated (not removed from disk, but they'll no longer be matched).
+#' However, if a file imported using [sass_file()] itself imports other sass
+#' files using \code{@import}, changes to those files are invisible to the cache
+#' and you will end up with stale results. Caching should be disabled in cases
+#' like this.
+#'
+#' If `sass_get_default_cache()` is called before `sass_set_default_cache()`,
+#' then it will create a cache in a subdirectory of the system temp directory
+#' named `R-sass-cache-username` and set it as the default. Because this cache
+#' directory is not in the R process's temp directory, it will persist longer
+#' than the R process, typically until a system reboot. Additionally, it will be
+#' shared across R processes for the same user on the same system.
+#'
+#' By default, the maximum size of the cache is 40 MB. If it grows past that
+#' size, the least-recently-used objects will be evicted from the cache to keep
+#' it under that size. To set the options for the cache, create a [FileCache]
+#' object and set it as the default cache.
+#'
+#' To clear the cache (but keep using it in the future), call
+#' `sass_get_default_cache()$reset()`.
+#'
+#' @param cache A [FileCache] object, or `NULL` if you don't want to use a
+#'   cache.
+#'
+#' @seealso sass_file_cache
+#'
+#' @examples
+#' # Very slow to compile
+#' fib_sass <- "@function fib($x) {
+#'   @if $x <= 1 {
+#'     @return $x
+#'   }
+#'   @return fib($x - 2) + fib($x - 1);
+#' }
+#'
+#' body {
+#'   width: fib(27);
+#' }"
+#'
+#' # The first time this runs it will be very slow
+#' system.time(sass(fib_sass))
+#'
+#' # But on subsequent calls, it should be very fast
+#' system.time(sass(fib_sass))
+#'
+#' # sass() can be called with cache=NULL; it will be slow
+#' system.time(sass(fib_sass, cache = NULL))
+#'
+#' # Clear the cache
+#' sass_get_default_cache()$reset()
+#'
+#' \dontrun{
+#' # Example of disabling cache by setting the default cache to NULL.
+#'
+#' # Disable the default cache (save the original one first, so we can restore)
+#' old_cache <- sass_get_default_cache()
+#' sass_set_default_cache(NULL)
+#' # Will be slow, because no cache
+#' system.time(sass(fib_sass))
+#'
+#' # Restore the original cache
+#' sass_set_default_cache(old_cache)
+#' }
+#' @export
+sass_get_default_cache <- function() {
+  if (!exists("cache", .globals, inherits = FALSE)) {
+    username <- Sys.info()[["user"]]
+    sass_set_default_cache(sass_file_cache())
+  }
+  .globals$cache
+}
+
+#' @rdname sass_get_default_cache
+#' @export
+sass_set_default_cache <- function(cache = sass_file_cache()) {
```

Comment:
Why not `sass_cache_set()`/`sass_cache_get()` instead of `sass_set_default_cache()`?

Reply:
Because users can still specify the cache when they call `sass()`. This is similar to getting/setting default themes with ggplot2. I don't feel strongly about it, though.

Reply:
Yea, I get that, I'm just mainly wondering why you decided to include `default` in the name (it seems unnecessary)

Reply:
A naming like `sass_cache_set()`/`sass_cache_get()` would be more consistent with the naming convention I've been using in bootstraplib and thematic (e.g., `bs_theme_set()`/`bs_theme_get()`).

Reply:
Done.